### PR TITLE
Add prism for refined types

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/util/NewType.scala [hunk 1]
```diff
@@ -10,6 +10,7 @@ import eu.timepit.refined.api.*
 import io.circe.Decoder
 import io.circe.Encoder
 import monocle.Iso
```
Comment (rpiaggio) [inline modules/core/shared/src/main/scala/lucuma/core/util/NewType.scala:12]:
```suggestion
import lucuma.core.optics.refinedPrism
import monocle.Iso
```

File: modules/core/shared/src/main/scala/lucuma/core/util/NewType.scala [hunk 2]
```diff
@@ -40,5 +41,6 @@ trait NewType[Wrapped]:
 
 trait RefinedNewType[T, P](using RefinedType.AuxT[T Refined P, T]) extends NewType[T Refined P]:
   private val TypeOps = new RefinedTypeOps[T Refined P, T]
+  def from: Prism[T, Type] = Prism[T, Type](from(_).toOption)(_.value.value)
```
Comment (rpiaggio) [inline modules/core/shared/src/main/scala/lucuma/core/util/NewType.scala:44]:
```suggestion
  def from(using Validate[T, P]): Prism[T, Type] = refinedPrism.andThen(value.reverse)```

**Review by rpiaggio — APPROVED** [review]
LGTM. The suggestion is a bit of nitpicking, but composing existing optics instead of building a new one guarantees lawfulness in the general case.

**Comment by cquiroz** [comment]
Great idea

**Pull request merged**